### DISCUSSION TITLE
Pinned Components 📌

### DISCUSSION THREAD

**Discussion by Thane5** [neon-age/Smart-Hierarchy#19] [Ideas]
Here's an idea i've had for a while. There is actually many older addons that attempt to do something similar, but it often ended up in a messy solution. I'm talking about displaying multiple icons in the Hierarchy window, to show more info on other components that are attached to the Object. Of course i could simply prioritize the script icons, but then i would loose most of the useful info about the object. Scripts, or in my case, Bolt Flowgraphs (Aka Unity visual scripting in the current Beta) are more of an additional, or secondary component with very generic icons that hold no information about its purpose. So why not show these icons after the Object name instead?
![Hierarchy Concept](https://user-images.githubusercontent.com/13683581/109296470-9f38df80-7830-11eb-8e96-fe981c213ea8.png)
Of course this should always stay an option, since it could quickly make the hierarchy window messy, but i think its very valuable to quickly see which object have scripts attached and which don't.

**Comment by neon-age** [comment]
Yes, I think it would be a good idea to always show "important" components after the name. Like Cinemachine does this with "brain" camera.

Actually, it's directly related to what I've been working on for awhile now - Components Toolbar:
![image](https://user-images.githubusercontent.com/29812914/109321150-67a75300-7883-11eb-9635-82a061dd2b18.png)
But it only shows all the attached components on hover/selection.

So, how should we decide which components are always shown? 
I'm thinking of extending Types Priority list with a "Pin" toggle:
![image](https://user-images.githubusercontent.com/29812914/109369745-2c7f4100-78d0-11eb-9d19-a455ee356880.png)


In a meantime, I've published components-toolbar branch, you can try it using this URL:
https://github.com/neon-age/Smart-Hierarchy.git#components-toolbar
